src/SprykerEco/Zed/PunchoutCatalogs/Business/Reader/PunchoutCatalogsReader.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\PunchoutCatalogs\Business\Reader;
+
+use Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer;
+use SprykerEco\Zed\PunchoutCatalogs\Dependency\Facade\PunchoutCatalogsToVaultFacadeInterface;
+use SprykerEco\Zed\PunchoutCatalogs\Persistence\PunchoutCatalogsRepositoryInterface;
+
+class PunchoutCatalogsReader implements PunchoutCatalogsReaderInterface
+{
+    protected const VAULT_DATA_TYPE_PASSWORD = 'pwg_punchout_catalog_connection.password';
+
+    /**
+     * @var \SprykerEco\Zed\PunchoutCatalogs\Persistence\PunchoutCatalogsRepositoryInterface
+     */
+    protected $punchoutCatalogsRepository;
+
+    /**
+     * @var \SprykerEco\Zed\PunchoutCatalogs\Dependency\Facade\PunchoutCatalogsToVaultFacadeInterface
+     */
+    protected $vaultFacade;
+
+    /**
+     * @param \SprykerEco\Zed\PunchoutCatalogs\Persistence\PunchoutCatalogsRepositoryInterface $punchoutCatalogsRepository
+     * @param \SprykerEco\Zed\PunchoutCatalogs\Dependency\Facade\PunchoutCatalogsToVaultFacadeInterface $vaultFacade
+     */
+    public function __construct(PunchoutCatalogsRepositoryInterface $punchoutCatalogsRepository, PunchoutCatalogsToVaultFacadeInterface $vaultFacade)
+    {
+        $this->punchoutCatalogsRepository = $punchoutCatalogsRepository;
+        $this->vaultFacade = $vaultFacade;
+    }
+
+    /**
+     * @param int $connectionId
+     *
+     * @return \Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer|null
+     */
+    public function findConnectionById(int $connectionId): ?PunchoutCatalogConnectionTransfer
+    {
+        $punchoutCatalogConnectionTransfer = $this->punchoutCatalogsRepository->findConnectionById($connectionId);
+
+        if (!$punchoutCatalogConnectionTransfer) {
+            return $punchoutCatalogConnectionTransfer;
+        }
+
+        $password = $this->vaultFacade->retrieve(
+            static::VAULT_DATA_TYPE_PASSWORD,
+            (string)$punchoutCatalogConnectionTransfer->getIdPunchoutCatalogConnection()
```

Comment:
1) Why ID Punchout Catalog Connection is a string?
2) Is this ID used as a data type for Vault or what is a second param?

Reply:
```
    public function store(string $dataType, string $dataKey, string $data): bool
    {
        return $this->getFactory()
            ->createVaultWriter()
            ->store($dataType, $dataKey, $data);
    }

```